### ISSUE TITLE
tf.name_scope does not obey its own documented rules

### ISSUE THREAD

**Issue by failure-to-thrive** [tensorflow/tensorflow#37264]
**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04.3 LTS (GNU/Linux 4.14.137+ x86_64)
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.1.0-0-ge5bf8de410
- Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)

\
According to the https://www.tensorflow.org/api_docs/python/tf/name_scope:

> If the scope name already exists, the name will be made unique by appending _n. For example, calling my_op the second time will generate MyOp_1/a, etc.

However, it turned out to be not. Nor did I manage to find any code (inside `tf.name_scope`) that might generate the uniqueness.

``` python
import tensorflow as tf
layer = tf.keras.Input(shape = [None])
def get_shape(layer):
  with tf.name_scope('scope') as scope:
    return tf.shape(layer, name=scope)
get_shape(layer)
get_shape(layer)
```
The first `get_shape(layer)`:
```
<tf.Tensor 'scope:0' shape=(2,) dtype=int32>
```
The second `get_shape(layer)`:
```
Traceback (most recent call last):
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/ops.py", line 1619, in _create_c_op
    c_op = c_api.TF_FinishOperation(op_desc)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Duplicate node name in graph: 'scope'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in get_shape
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/array_ops.py", line 519, in shape_v2
    return shape(input, name, out_type)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/array_ops.py", line 545, in shape
    return shape_internal(input, name, optimize=True, out_type=out_type)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/array_ops.py", line 573, in shape_internal
    return gen_array_ops.shape(input, name=name, out_type=out_type)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/gen_array_ops.py", line 8234, in shape
    "Shape", input=input, out_type=out_type, name=name)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/op_def_library.py", line 742, in _apply_op_helper
    attrs=attr_protos, op_def=op_def)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/func_graph.py", line 595, in _create_op_internal
    compute_device)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/ops.py", line 3322, in _create_op_internal
    op_def=op_def)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/ops.py", line 1786, in __init__
    control_input_ops)
  File "/tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/ops.py", line 1622, in _create_c_op
    raise ValueError(str(e))
ValueError: Duplicate node name in graph: 'scope'
```

**Comment by Saduf2019** [comment]
@failure-to-thrive
I have tried to replicate your issue and observe that the error is reproducible even for single use of get_shape,please refer to the [gist](https://colab.sandbox.google.com/gist/Saduf2019/9096209a8e0752e0bb96d08a727635e7/37264.ipynb) and let us know if that helps.
please refer to this link for [usage of shape](https://www.tensorflow.org/api_docs/python/tf/shape?hl=sv)

**Comment by failure-to-thrive** [comment]
> I have tried to replicate your issue and observe that the error is reproducible even for single use of get_shape,please refer to the [gist](https://colab.sandbox.google.com/gist/Saduf2019/9096209a8e0752e0bb96d08a727635e7/37264.ipynb) and let us know if that helps.

The reason why you may observe the error even after the first use - you already launched the script and haven't noticed that. You should reset the notebook. The 'scope' is in the graph and will not go away on its own.

> please refer to this link for [usage of shape](https://www.tensorflow.org/api_docs/python/tf/shape?hl=sv)

Thanks, but my point is to demonstrate that the `tf.name_scope` doesn't work as documented. `tf.shape` has nothing to do with that. It's just for an example. You may try `print(scope.name)`

**Comment by ymodak** [comment]
This is fixed with tf-nightly version '2.2.0-dev20200309'. Thanks!
